pybamm/solvers/base_solver.py

```diff
@@ -573,22 +600,30 @@ def solve(self, model, t_eval=None, external_variables=None, inputs=None):
                 # If the new initial conditions are different, set up again
                 # Doing the whole setup again might be slow, but no need to prematurely
                 # optimize this
-                self.set_up(model, ext_and_inputs, t_eval)
+                self.set_up(model, ext_and_inputs_list[0], t_eval)
                 self.models_set_up[model][
                     "initial conditions"
                 ] = model.concatenated_initial_conditions
         set_up_time = timer.time()
         timer.reset()
 
         # (Re-)calculate consistent initial conditions
-        self._set_initial_conditions(model, ext_and_inputs, update_rhs=True)
+        # Assuming initial conditions do not depend on input parameters
+        # when len(inputs_list) > 1, only `ext_and_inputs_list[0]`
+        # is passed to `_set_initial_conditions`.
+        # See https://github.com/pybamm-team/PyBaMM/pull/1261
+        self._set_initial_conditions(model, ext_and_inputs_list[0], update_rhs=True)
```

Comment:
There should be an error check for this. You can check what input parameters are in `model.concatenated_initial_conditions.pre_order()` and make sure none of them appear in ext_and_inputs_list.

Reply:
Done!